docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/spryker-elastic-computing.md

```diff
@@ -0,0 +1,316 @@
+---
+title: Spryker Elastic Computing
+description: This guideline explains how to use Elastic Computing functionality.
+last_updated: May 16, 2022
+template: concept-topic-template
+---
+
+Spryker is shipped with several elastic computing features that let you to utilize computational resources more efficiently. 
+
+## Monitoring
+
+To enable New Relic monitoring for P&S and the infastructure of queue workers, follow the steps:
+
+1. Update the `spryker/monitoring` module to version 2.5.0 or higher.
+2. In `Pyz\Zed\Monitoring\MonitoringConfig`, add `queue:task:start` command to the argument grouped transactions:
+```php
+namespace Pyz\Zed\Monitoring;
+
+use Spryker\Zed\Monitoring\MonitoringConfig as BaseMonitoringConfig;
+
+class MonitoringConfig extends BaseMonitoringConfig
+{
+    /**
+     * @api
+     *
+     * @return array<string>
+     */
+    public function getGroupedByArgumentTransactions(): array
+    {
+        return ['queue:task:start'];
+    }
+}
+```
+
+3. Add to the `Pyz\Zed\Monitoring\Business\MonitoringBusinessFactory` following lines of code:
+```php
+public function getMonitoringTransactionNamingStrategies(): array
+    {
+        return [
+            return new FirstArgumentMonitoringConsoleTransactionNamingStrategy($this->getConfig()->getGroupedByArgumentTransactions());
+        ];
+    }
+```
+This will aggregate all different queue worker transactions under corresponding `queue:task:start {queue_name}` transaction name in New Relic.
```

Comment:
```suggestion
Now New Relic aggregates all queue worker transactions under the name specified in `queue:task:start {queue_name}`.
```

Reply:
Now New Relic transactions of `queue:task:start` are aggregated under separate names for each queue name.